internal/datasourcev2/polling_data_source.go

```diff
@@ -0,0 +1,201 @@
+package datasourcev2
```

Comment:
As last time, most of this is a straight copy from the older `datasource` package. The only changes are in the requester, so theoretically we could use all the existing stuff and just replace that piece.

However, since we are planning to evolve this work pretty quickly, I figured the duplication is likely preferable.

I will call out where the actual changes occur.